src/components/instantlaunches/InstantLaunchButtonWrapper.js

```diff
@@ -51,21 +55,30 @@ function InstantLaunchButtonWrapper(props) {
         React.useState(false);
     const [runErrorDetails, setRunErrorDetails] = React.useState(null);
     const [ilUrl, setIlUrl] = React.useState();
+    const [landingUrl, setLandingUrl] = React.useState();
 
+    const router = useRouter();
     const { t } = useTranslation("launch");
 
     React.useEffect(() => {
-        if (ilUrl) {
+        if (ilUrl && landingUrl) {
```

Comment:
Because the landingUrl getting set depends on even less than the ilUrl, I think it's safe to have this be a straight `&&` rather than also guarding on the value of `autolaunch`, but it could be changed to `ilUrl && (!autolaunch || landingUrl)` if we'd rather be more careful, I guess.

Reply:
LGTM. If fact, this could have remained just `if (ilUrl)` and below could check `if (autolaunch && landingUrl)`.